kafka/src/main/java/org/apache/calcite/adapter/kafka/KafkaTableFactory.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.kafka;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.TableFactory;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+/**
+ * Implementation of {@link TableFactory} for Kafka. Currently a Kafka topic is mapping
+ * to a STREAM table.
+ */
+public class KafkaTableFactory implements TableFactory {
+  public KafkaTableFactory() {
+  }
+
+  /** Creates a Table.
+   *  @param schema Schema this table belongs to
+   * @param name Name of this table
+   * @param operand The "operand" JSON property
+   * @param rowType Row type. Specified if the "columns" JSON property.
+   */
+  @Override public Table create(final SchemaPlus schema, final String name, final Map operand,
+      final RelDataType rowType) {
+    KafkaTableOptions.Builder tableOptionBuilder = KafkaTableOptions.builder();
+
+    tableOptionBuilder.bootstrapServers(
+        (String) operand.getOrDefault(KafkaTableConstants.SCHEMA_BOOTSTRAP_SERVERS, null));
+    tableOptionBuilder.topicName(
+        (String) operand.getOrDefault(KafkaTableConstants.SCHEMA_TOPIC_NAME, null));
+
+    KafkaRowConverter rowConverter = null;
+    if (operand.containsKey(KafkaTableConstants.SCHEMA_ROW_CONVERTER)) {
+      try {
+        rowConverter = (KafkaRowConverter) Class.forName(
+            (String) operand.get(KafkaTableConstants.SCHEMA_ROW_CONVERTER))
+            .newInstance();
+      } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+        throw new RuntimeException(e);
```

Comment:
Can you give more details in exception message (eg. `Failed to create table $T in schema $S`)?  In current context you have table name and config.

Reply:
updated with detailed error message